src/stream/src/executor/mview/materialize.rs

```diff
@@ -83,7 +84,18 @@ impl<S: StateStore, SD: ValueRowSerde> MaterializeExecutor<S, SD> {
     ) -> Self {
         let arrange_key_indices: Vec<usize> = arrange_key.iter().map(|k| k.column_index).collect();
 
-        let state_table = StateTableInner::from_table_catalog(table_catalog, store, vnodes).await;
+        let can_disable_conflict_check = actor_context.dispatch_num == 0;
+
+        let state_table = if matches!(
+            conflict_behavior,
+            ConflictBehavior::Overwrite | ConflictBehavior::IgnoreConflict
+        ) && can_disable_conflict_check
+        {
+            // Table could disable conflict check if no downstream mv depends on it, so we use a inconsistent_op to skip sanity check as well.
+            StateTableInner::from_table_catalog_inconsistent_op(table_catalog, store, vnodes).await
```

Comment:
When implementing partial checkpoint, we need to leverage the consistent op to store the correct value. Therefore, I think the storage side also be aware of the switch between `disable_conflict_check` or not. 

In the `commit` of `StateTable` we shall support enabling or disabling the consistent op sanity check at runtime, and in the `seal_current_epoch` of `LocalStateStore` we can add a flag in `SealCurrentEpochOptions` to optionally change the `is_consistent_op` value.

Reply:
> In the commit of StateTable we shall support enabling or disabling the consistent op sanity check at runtime,

Currently, the state table flag `inconsistent_op` switches only when the actor is rebuilt. You said at runtime, do you mean when a new downstream mv is created, we need to tell the storage?

Reply:
Yes. It means we can shall support switching between `inconsistent_op` and `consistent_op` when the value of `can_disable_overwrite_conflict_check` is changed without rebuilding the actor.